### PR TITLE
Minor improvements to address code quality issues reported on dev branch

### PR DESCRIPTION
**Pull request by thomasplevy**
See errors reported on #1437 at https://codeclimate.com/github/gocodebox/lifterlms/pull/1437

This change seeks to address these issues (while adding more test coverage)

### CODE REVIEW

**Comment by eri-trabiccolo** [comment]
everybody disliked that piece of code but it was necessary...

File: includes/llms.functions.core.php [hunk 1]
```diff
@@ -511,6 +511,40 @@ function llms_get_enrollable_status_check_post_types() {
...
+		E_CORE_WARNING      => 'E_CORE_WARNING', // 32.
+		E_COMPILE_ERROR     => 'E_COMPILE_ERROR', // 64.
+		E_COMPILE_WARNING   => 'E_COMPILE_WARNING', // 128.
+		E_USER_ERROR        => 'E_USER_ERROR', // 256.
+		E_USER_WARNING      => 'E_USER_WARNING', // 512.
+		E_USER_NOTICE       => 'E_USER_NOTICE', // 1024.
+		E_STRICT            => 'E_STRICT', // 2048.
+		E_RECOVERABLE_ERROR => 'E_RECOVERABLE_ERROR', // 4096.
+		E_DEPRECATED        => 'E_DEPRECATED', // 8192.
+		E_USER_DEPRECATED   => 'E_USER_DEPRECATED',   // 16384.
```
Comment (eri-trabiccolo) [inline includes/llms.functions.core.php]:
```suggestion
		E_USER_DEPRECATED   => 'E_USER_DEPRECATED',  // 16384.
```

File: includes/class.llms.generator.php [hunk 2]
```diff
@@ -162,43 +162,28 @@ public function generate() {
...
-			case E_STRICT: // 2048.
-				return 'E_STRICT';
-			case E_RECOVERABLE_ERROR: // 4096.
-				return 'E_RECOVERABLE_ERROR';
-			case E_DEPRECATED: // 8192.
-				return 'E_DEPRECATED';
-			case E_USER_DEPRECATED: // 16384.
-				return 'E_USER_DEPRECATED';
+		// See if the error code is a native php exception code constant.
+		$ret = llms_error_constant_to_code( $code );
```
Comment (eri-trabiccolo) [inline includes/class.llms.generator.php]: shall we use a more meaningful name here so that it's more clear we only map native PHP error constants?!

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
Feel free to merge it anyways, nothing critical, fine for me.

File: includes/class.llms.generator.php (hunk 2 above)
Comment (thomasplevy) [reply includes/class.llms.generator.php]: you mean a more meaningful function name? what do you propose?

Comment (eri-trabiccolo) [reply includes/class.llms.generator.php]: - [ ] `llms_php_native_error_constant_to_code`
- [ ] `llms_native_error_constant_to_code`

Comment (thomasplevy) [reply includes/class.llms.generator.php]: How about just `llms_php_error_constant_to_code` ?

Comment (thomasplevy) [reply includes/class.llms.generator.php]: (Yours are fine if you're not into the whole brevity thing)

Comment (eri-trabiccolo) [reply includes/class.llms.generator.php]: ok, fine enough <3

**Pull request merged**